web/src/elements/forms/form-associated-element.ts

```diff
@@ -0,0 +1,216 @@
+import { AKElement } from "#elements/Base";
+
+import { Jsonifiable } from "type-fest";
+
+import { msg } from "@lit/localize";
+import { LitElement } from "lit";
+import { property } from "lit/decorators.js";
+import { createRef, Ref } from "lit/directives/ref.js";
+
+/**
+ * A subset of form associated {@linkcode ElementInternals} properties.
+ *
+ * @see {@linkcode FormAssociatedElement} for usage.
+ */
+export interface FormAssociated
```

Comment:
This is largely ported from the first iteration in #16010 

I've gone back and forth on whether this is better off as a mixin -- my current thought is that it would be level of indirection that could make debugging the internals object a challenge. 

However, if we do want a mixin in the future, we can incorporate this base class instead of dynamically creating one like we usually do 😊